connection_test.go

```diff
@@ -82,3 +83,193 @@ func TestDoubleClose(t *testing.T) {
 		t.Errorf("did not shutdown within timeout")
 	}
 }
+
+// TestHandleConnectionError_ProtocolsDone tests that connection errors are ignored
+// when main protocols are explicitly stopped
+func TestHandleConnectionError_ProtocolsDone(t *testing.T) {
+	defer goleak.VerifyNone(t)
+
+	mockConn := ouroboros_mock.NewConnection(
+		ouroboros_mock.ProtocolRoleClient,
+		[]ouroboros_mock.ConversationEntry{
+			ouroboros_mock.ConversationEntryHandshakeRequestGeneric,
+			ouroboros_mock.ConversationEntryHandshakeNtNResponse,
+		},
+	)
+
+	oConn, err := ouroboros.New(
+		ouroboros.WithConnection(mockConn),
+		ouroboros.WithNetworkMagic(ouroboros_mock.MockNetworkMagic),
+		ouroboros.WithNodeToNode(true),
+	)
+	if err != nil {
+		t.Fatalf("unexpected error when creating Connection object: %s", err)
+	}
+
+	// Test through HandleConnectionError - should return error when protocols are active
+	testErr := io.EOF
+	err = oConn.HandleConnectionError(testErr)
+	if err != testErr {
+		t.Fatalf("expected original error when protocols are active, got: %v", err)
+	}
```

Comment:
This isn't really testing anything useful. Instead, you should start one of the protocols and then close the connection (both from the "remote" side) and see if you get the expected error(s).